3rd-party/class.jetpack-amp-support.php

```diff
@@ -0,0 +1,320 @@
+		add_filter( 'jp_carousel_maybe_disable', '__return_true' );
+
+		// sharing
+		add_filter( 'sharing_enqueue_scripts', '__return_false' );
+		add_filter( 'jetpack_sharing_counts', '__return_false' );
+		add_filter( 'sharing_js', '__return_false' );
+		add_filter( 'jetpack_sharing_display_markup', array( 'Jetpack_AMP_Support', 'render_sharing_html' ), 10, 2 );
+
+		// disable lazy images
+		add_filter( 'lazyload_is_enabled', '__return_false' );
+
+		// disable imploding CSS
+		add_filter( 'jetpack_implode_frontend_css', '__return_false' );
+
+		// enforce freedom mode for videopress
+		add_filter( 'videopress_shortcode_options', array( 'Jetpack_AMP_Support', 'videopress_enable_freedom_mode' ) );
+
+		// include Jetpack og tags when rendering native AMP head
+		add_action( 'amp_post_template_head', array( 'Jetpack_AMP_Support', 'amp_post_jetpack_og_tags' ) );
+
+		// Post rendering changes for legacy AMP
+		add_action( 'pre_amp_render_post', array( 'Jetpack_AMP_Support', 'amp_disable_the_content_filters' ) );
+
+		// Add post template metadata for legacy AMP
+		add_filter( 'amp_post_template_metadata', array( 'Jetpack_AMP_Support', 'amp_post_template_metadata' ), 10, 2 );
+	}
+
+	static function admin_init() {
+		// disable Likes metabox for post editor if AMP canonical disabled
+		add_filter( 'post_flair_disable',  array( 'Jetpack_AMP_Support', 'is_amp_canonical' ), 99 );
+	}
+
+	static function init_filter_jetpack_widgets() {
+		if ( ! self::is_amp_request() ) {
+			return;
+		}
+
+		// widgets
+		add_filter( 'jetpack_widgets_to_include', array( 'Jetpack_AMP_Support', 'filter_available_widgets' ) );
+	}
+
+	static function is_amp_canonical() {
+		return function_exists( 'amp_is_canonical' ) && amp_is_canonical();
+	}
+
+	static function is_amp_request() {
+		// can't use is_amp_endpoint() since it's not ready early enough in init.
+		// is_amp_endpoint() implementation calls is_feed, which bails with a notice if plugins_loaded isn't finished
+		// "Conditional query tags do not work before the query is run"
+		$is_amp_request = ! is_admin() // this is necessary so that modules can still be enabled/disabled/configured as per normal via Jetpack admin
+			&&
+				function_exists( 'amp_is_canonical' ) // this is really just testing if the plugin exists
+			&&
+				( amp_is_canonical() || isset( $_GET[ amp_get_slug() ] ) );
+
+		/**
+		 * Returns true if the current request should return valid AMP content.
+		 *
+		 * @since 6.2.0
+		 *
+		 * @param boolean $is_amp_request Is this request supposed to return valid AMP content?
+		 */
+		return apply_filters( 'jetpack_is_amp_request', $is_amp_request );
+	}
+
+	static function filter_available_widgets( $widgets ) {
+		if ( self::is_amp_request() ) {
+			$widgets = array_filter( $widgets, array( 'Jetpack_AMP_Support', 'is_supported_widget' ) );
+		}
+
+		return $widgets;
+	}
+
+	static function is_supported_widget( $widget_path ) {
+		return substr($widget_path, -14) !== '/milestone.php';
```

Comment:
Minor style nitpick: There should be spaces inside the parentheses.

Reply:
Fixed in new PR